src/java/main/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -817,10 +817,12 @@ protected void pRequest(Request request) throws RequestProcessorException {
                             type = OpCode.error;
                             txn = new ErrorTxn(e.code().intValue());
 
-                            LOG.info("Got user-level KeeperException when processing "
-                                    + request.toString() + " aborting remaining multi ops."
-                                    + " Error Path:" + e.getPath()
-                                    + " Error:" + e.getMessage());
+                            if (e.code().intValue() > Code.APIERROR.intValue()) {
```

Comment:
Does this mean that we don't want to log system errors at all?
Or I might be missing something here.

Reply:
Additionally refactoring this one to use message formatter would be nice.

Reply:
this one really seems like a debug level to me. the issue is that the client request tried to do something create a node that already exists. these are very normal cases that the client is probably expecting (in the case of leader election, for example). we shouldn't be logging this kind of expected behavior for common operations as INFO

Reply:
I get the point - things will break when they are running at scale :) . I don't have strong opinion holding this as INFO. wondering how others think about this (I mentioned it's info as this log was my good friend before when I debugged some issues...).

BTW for similar use case (large scale of clients) - it looks like all client side related log should be silenced on server side. basically server side just keep server specific loggings on by default (which hopefully does not change regardless of the scale of clients). So all client cnx / session related logs need be silenced...

Reply:
Exactly.

I'll fix up the format of this log line and the others as well.